firmware/ventilator-controller-stm32/Core/Src/Pufferfish/Driver/I2C/LTC4015/Sensor.cpp

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2020, the Pez Globo team and the Pufferfish project contributors
+ *
+ *      Author: Rohan Purohit
+ *
+ */
+
+#include "Pufferfish/Driver/I2C/LTC4015/Sensor.h"
+
+#include "Pufferfish/Util/Timeouts.h"
+
+namespace Pufferfish::Driver::I2C::LTC4015 {
+
+// StateMachine
+
+StateMachine::Action StateMachine::update(uint32_t current_time_us) {
+  current_time_us_ = current_time_us;
+  switch (next_action_) {
+    case Action::initialize:
+      next_action_ = Action::measure;
+      break;
+    case Action::measure:
+      break;
+  }
+  return next_action_;
+}
+
+// Sensor
+
+InitializableState Sensor::setup() {
+  switch (next_action_) {
+    case Action::initialize:
+      return initialize(time_.micros());
+    case Action::measure:
+      return InitializableState::ok;
+  }
+  return InitializableState::failed;
+}
+
+InitializableState Sensor::output(PowerManagement &power_management) {
+  switch (next_action_) {
+    default:
```

Comment:
Since you put the `default` case up here, I think you'll always fall through to `case Action::measure` and return from inside it, and you'll never reach the `return InitializableState::failed` line at the end of this method. Maybe you meant to move the `default` case to the end of the `switch` statement?

Reply:
Yes fixed!